crates/services/txpool_v2/src/extracted_outputs.rs

```diff
@@ -0,0 +1,129 @@
+
+use std::collections::HashMap;
+
+use fuel_core_types::{
+    fuel_tx::{
+        input::coin::{
+            CoinPredicate,
+            CoinSigned,
+        },
+        Address,
+        AssetId,
+        ContractId,
+        Input,
+        Output,
+        TxId,
+        UtxoId,
+    },
+    services::txpool::ArcPoolTx,
+};
+
+pub struct ExtractedOutputs {
+    contract_created: HashMap<ContractId, TxId>,
+    contract_created_by_tx: HashMap<TxId, Vec<ContractId>>,
+    coins_created: HashMap<TxId, HashMap<u16, (Address, u64, AssetId)>>,
+}
+
+impl ExtractedOutputs {
+    pub fn new() -> Self {
+        Self {
+            contract_created: HashMap::new(),
+            contract_created_by_tx: HashMap::new(),
+            coins_created: HashMap::new(),
+        }
+    }
+}
+
+impl Default for ExtractedOutputs {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ExtractedOutputs {
+    pub fn new_extracted_transaction(&mut self, tx: &ArcPoolTx) {
+        let tx_id = tx.id();
+        for (idx, output) in tx.outputs().iter().enumerate() {
+            match output {
+                Output::ContractCreated { contract_id, .. } => {
+                    self.contract_created.insert(*contract_id, tx_id);
+                    self.contract_created_by_tx
+                        .entry(tx_id)
+                        .or_default()
+                        .push(*contract_id);
+                }
+                Output::Coin {
+                    to,
+                    amount,
+                    asset_id,
+                } => {
+                    self.coins_created.entry(tx_id).or_default().insert(
+                        u16::try_from(idx).expect("Outputs count is less than u16::MAX"),
+                        (*to, *amount, *asset_id),
+                    );
+                }
+                Output::Contract { .. }
+                | Output::Change { .. }
+                | Output::Variable { .. } => {
+                    continue;
+                }
+            }
+        }
+        for input in tx.inputs() {
+            match input {
+                Input::CoinSigned(CoinSigned { utxo_id, .. })
+                | Input::CoinPredicate(CoinPredicate { utxo_id, .. }) => {
+                    self.coins_created
+                        .entry(*utxo_id.tx_id())
+                        .and_modify(|coins| {
+                            coins.remove(&utxo_id.output_index());
+                        });
+                }
+                Input::Contract(_)
+                | Input::MessageCoinPredicate(_)
+                | Input::MessageCoinSigned(_)
+                | Input::MessageDataPredicate(_)
+                | Input::MessageDataSigned(_) => {
+                    continue;
+                }
+            }
+        }
+    }
+
+    pub fn new_skipped_transaction(&mut self, tx_id: &TxId) {
+        self.new_executed_transaction(tx_id);
+    }
```

Comment:
`new_skipped_transaction()` just calling `new_executed_transaction` is confusing when you don't know the context. Maybe some renaming or at least additional comment could help

At first glance this code is similar to something like this :smile: 
```
    fn insert(&self, val) { self.remove(val) }
```